src/fswAlgorithms/effectorInterfaces/solarArrayReference/solarArrayReference.c

```diff
@@ -69,5 +71,98 @@ void Reset_solarArrayReference(solarArrayReferenceConfig *configData, uint64_t c
 */
 void Update_solarArrayReference(solarArrayReferenceConfig *configData, uint64_t callTime, int64_t moduleID)
 {
-    
+     /*! - Create and assign buffer messages */
+    NavAttMsgPayload            attNavIn = NavAttMsg_C_read(&configData->attNavInMsg);
+    AttRefMsgPayload            attRefIn = AttRefMsg_C_read(&configData->attRefInMsg);
+    HingedRigidBodyMsgPayload   hingedRigidBodyIn     = HingedRigidBodyMsg_C_read(&configData->hingedRigidBodyInMsg);
+    HingedRigidBodyMsgPayload   hingedRigidBodyRefOut = HingedRigidBodyMsg_C_zeroMsgPayload();
+
+    /*! read Sun direction in B frame from the attNav message and map it to R frame */
+    double rHat_SB_B[3];    // Sun direction in body-frame coordinates
+    double rHat_SB_R[3];    // Sun direction in reference-frame coordinates
+    double BN[3][3];   // inertial to body frame DCM
+    double RN[3][3];   // inertial to reference frame DCM
+    double RB[3][3];   // body to reference DCM
+    v3Normalize(attNavIn.vehSunPntBdy, rHat_SB_B);
+    switch (configData->bodyFrame) {
+
+        case 0:
+        MRP2C(attNavIn.sigma_BN, BN);
+        MRP2C(attRefIn.sigma_RN, RN);
+        m33MultM33t(RN, BN, RB);
+        m33MultV3(RB, rHat_SB_B, rHat_SB_R);
+        break;
+
+        case 1:
+        v3Copy(rHat_SB_B, rHat_SB_R);
+        break;
+
+        default:
+        _bskLog(configData->bskLogger, BSK_ERROR, "Error: solarArrayAngle.bodyFrame input can be either 0 or 1.");
+    }
+
+    /*! compute solar array frame axes at zero rotation */
+    double a1Hat_B[3];      // solar array axis drive
+    double a2Hat_B[3];      // solar array axis surface normal
+    double a3Hat_B[3];      // third axis according to right-hand rule
+    v3Normalize(configData->a1Hat_B, a1Hat_B);
+    v3Cross(a1Hat_B, configData->a2Hat_B, a3Hat_B);
+    v3Normalize(a3Hat_B, a3Hat_B);
+    v3Cross(a3Hat_B, a1Hat_B, a2Hat_B);
+
+    /*! compute solar array reference frame axes at zero rotation */
+    double a1Hat_R[3];
+    double a2Hat_R[3];
+    double dotP = v3Dot(a1Hat_B, rHat_SB_R);
+    for (int n = 0; n < 3; n++) {
+        a2Hat_R[n] = rHat_SB_R[n] - dotP * a1Hat_B[n];
+    }
+    v3Normalize(a2Hat_R, a2Hat_R);
+    v3Cross(a2Hat_B, a2Hat_R, a1Hat_R);
+
+    /*! compute current rotation angle thetaC from input msg */
+    double sinThetaC = sin(hingedRigidBodyIn.theta);
+    double cosThetaC = cos(hingedRigidBodyIn.theta);
+    double thetaC = atan2(sinThetaC, cosThetaC);      // clip theta current between 0 and 2*pi
+
+    /*! compute reference angle thetaR and store in buffer msg */
+    double thetaR;
```

Comment:
:small_red_triangle_down: This variable needs to be initialized. It could be used uninitialized on line 159 and following on 162.

Reply:
Are you talking about thetaR? Not sure what you mean here. Also, since I pushed the changes requested by Dr. Schaub, the lines have shifted and I don't know what your were referring to anymore

Reply:
You are right. I'm referring to `thetaR`.

Reply:
I made some modifications, but I'm unsure whether that's what you meant. Please confirm if the current changes address your comment.

Reply:
I agree with the modification Riccardo did on `thetaR`.  Should be good to go now.